src/ir/effects.h

```diff
@@ -27,15 +27,22 @@ namespace wasm {
 
 struct EffectAnalyzer
   : public PostWalker<EffectAnalyzer, OverriddenVisitor<EffectAnalyzer>> {
-  EffectAnalyzer(const PassOptions& passOptions,
+  EffectAnalyzer(bool ignoreImplicitTraps,
```

Comment:
Do we need this separate constructor given that this is only used in the other constructor below?

Reply:
Typical usage here is the old constructor accepting `PassOptions`, except in `ExpressionRunner` where we do not know `PassOptions` but can make reasonable assumptions, which the new constructor is for. Shall I refactor the instantiations previously providing just `PassOptions` and make it use the new constructor instead, so we can remove the old one?

Reply:
Oh sorry, I missed you actually used the new constructor in your code. Nevermind. :)